specs/CDPSessionIdSupport.md

```diff
@@ -0,0 +1,328 @@
+  
+    wil::com_ptr<ICoreWebView2DevToolsProtocolEventReceiver> receiver;
+    // Listen to Runtime.consoleAPICalled event which is triggered when console.log is called by script code.
+    CHECK_FAILURE(
+        m_webView->GetDevToolsProtocolEventReceiver(L"Runtime.consoleAPICalled", &receiver));
+    CHECK_FAILURE(receiver->add_DevToolsProtocolEventReceived(
+        Callback<ICoreWebView2DevToolsProtocolEventReceivedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2DevToolsProtocolEventReceivedEventArgs* args) -> HRESULT
+            {
+                // Get console.log message details and which target it comes from
+                wil::unique_cotaskmem_string parameterObjectAsJson;
+                CHECK_FAILURE(args->get_ParameterObjectAsJson(&parameterObjectAsJson));
+                std::wstring eventSource;
+                std::wstring eventDetails = parameterObjectAsJson.get();
+                wil::com_ptr<ICoreWebView2DevToolsProtocolEventReceivedEventArgs2> args2;
+                if (SUCCEEDED(args->QueryInterface(IID_PPV_ARGS(&args2))))
+                {
+                    wil::unique_cotaskmem_string sessionId;
+                    CHECK_FAILURE(args2->get_SessionId(&sessionId));
+                    if (sessionId.get() && *sessionId.get())
+                    {
+                        std::wstring targetId = m_devToolsSessionMap[sessionId.get()];
+                        eventSource = m_devToolsTargetDescriptionMap[targetId];
+                    }
+                }
+                // App code to log these events.
+                LogConsoleLogMessage(eventSource, eventDetails);
+                return S_OK;
+            })
+            .Get(),
+        &m_consoleAPICalledToken));
+    receiver.reset();
+  
+    // Track Target and session info via CDP events.
+    CHECK_FAILURE(
+        m_webView->GetDevToolsProtocolEventReceiver(L"Target.attachedToTarget", &receiver));
+    CHECK_FAILURE(receiver->add_DevToolsProtocolEventReceived(
+        Callback<ICoreWebView2DevToolsProtocolEventReceivedEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2DevToolsProtocolEventReceivedEventArgs* args) -> HRESULT
+            {
+                // A new target is attached, add its info to maps.
+                wil::unique_cotaskmem_string jsonMessage;
+                CHECK_FAILURE(args->get_ParameterObjectAsJson(&jsonMessage));
+                std::wstring sessionId = GetJSONStringField(jsonMessage.get(), L"sessionId");
+                std::wstring targetId = GetJSONStringField(jsonMessage.get(), L"targetId");
+                m_devToolsSessionMap[sessionId] = targetId;
+                std::wstring type = GetJSONStringField(jsonMessage.get(), L"type");
+                std::wstring url = GetJSONStringField(jsonMessage.get(), L"url");
+                m_devToolsTargetDescriptionMap[targetId] = type + L"," + url;
```

Comment:
Nitpick: `insert_or_assign` is more efficient for setting elements into maps, since it emplaces the new string rather than creating an empty string and then overwriting it

Reply:
Please update